runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/AbstractDokkaParentTask.kt

```diff
@@ -2,81 +2,104 @@ package org.jetbrains.dokka.gradle
 
 import org.gradle.api.Project
 import org.gradle.api.Task
-import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.Nested
 import org.gradle.work.DisableCachingByDefault
 
+private const val DEPRECATION_MESSAGE = """
+    It is an anti-pattern to declare cross-project dependencies as it leads to various build problems. 
+    For this reason, this API wil be removed with the introduction of project isolation. 
+    As an alternative, don't apply the Dokka plugin in subprojects that you don't want to document, 
+    or disable individual tasks using the Gradle API.
```

Comment:
After some internal feedback, it looks like it can be helpful to mention that it's safe to use this API until project isolation comes along, and that we will provide a migration guide when it happens.

```suggestion
    It is an anti-pattern to declare cross-project dependencies as it leads to various build problems. 
    For this reason, this API wil be removed with the introduction of project isolation. 
    When it happens, we will provide a migration guide. In the meantime, you can keep using this API
    if you have to, but please don't rely on it if possible. If you don't want to document a certain project,
    don't apply the Dokka plugin for it, or disable individual project tasks using the Gradle API .
```